gradio/templates/colab.html

```diff
@@ -0,0 +1,26 @@
+<script>
+    document.querySelector("base").href = "https://localhost:$PORT";
+
+    function rehydrate() {
+        for (const script of document.querySelectorAll("script")) {
+            const newScript = document.createElement('script')
+            newScript.type = script.type;
+            newScript.src = script.src;
+            newScript.textContent = script.textContent;
+            const head = document.getElementsByTagName('head')[0];
+            if (script.textContent.includes("window.gradio_config =")) {
+                window.gradio_config = JSON.parse(script.textContent.replace("window.gradio_config =", "").replace(/;$/, "").trim());
+            }
```

Comment:
```suggestion
            if (script.textContent.includes("window.gradio_config =")) {
                const config_re = /^\s*window.gradio_config\s*=\s*({[^]+})\s*;\s*$/;
                window.gradio_config = JSON.parse(config_re.exec(script.textContent)[1]);            }
```

I think this regex is a bit more robust (slightly). 

I'm not entirely sure why the config isn't getting set correctly but it could be to do with different window scopes? or maybe there is something blocking it. I don't know if there is another solution but this does fix it.

Reply:
instead of parsing the js code, we code also get the config via GET request at `/config`? it requires an extra request but its more robust

Reply:
Yeah, we can do. I just figured it would be quite a bit slower. I'm happy with either, I only did this because it was already in the document.